torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -4063,7 +4063,11 @@ def _test_ddp_hook_parity(self, state, hook):
                 device_ids=[rank],
                 process_group=process_group,
             )
-            net_with_hook.register_comm_hook(state=state, hook=hook)
+            try:
```

Comment:
No need to update here if we don't need `test_ddp_hook_parity_allreduce_cpp_hook` below.